src/main/java/de/vogel612/testclient_javabot/TestProgram.java

```diff
@@ -39,14 +39,23 @@
 
 	public TestProgram() {
 		LOGGER.info("instantiating TestProgram class");
-		AppContext.INSTANCE.add(chatInterface);
-		bot = new DefaultBot(chatInterface);
-		gui = new ClientGui();
-		chatInterface.subscribe(bot);
-		chatInterface.subscribe(gui);
-		LOGGER.info("Basic component setup completed, beginning command glueing.");
-		new CoreBotCommands(chatInterface, bot).allCommands().forEach(bot::subscribe);
-		LOGGER.info("Glued Core Commands");
+        try{
+	        AppContext.INSTANCE.add(chatInterface);
+	        bot = new DefaultBot(chatInterface);
+	        Executors.newSingleThreadExecutor().execute(() -> {
```

Comment:
@TheItachiUchiha You are creating another `Executor` on this line without keeping any reference to it. If you would keep a reference to it, then you could shut it down in `Application.stop()`. (Note that in my fork, I removed this Executor entirely - and I also noted that other threads were not shutting down)